packages/frontend/src/app/submissions/page.tsx

```diff
@@ -0,0 +1,133 @@
+'use client'
+import React, { useEffect, useState } from 'react'
+import Image from 'next/image'
+import dayjs from 'dayjs'
+import { PageHeading } from '@/common/components/PageHeading'
+import { Text4xlBold } from '@/common/components/Typography'
+import { Submission } from '@/common/interfaces'
+import { PageContainer } from '@/common/components/PageContainer'
+import { Table, TableBody, TableHeader } from '@/common/components/Table'
+import { Filter, FilterButton } from '@/common/components/Filter'
+import { betterFetch } from '@/api'
+
+export default function Page() {
+  const [submissions, setSubmissions] = useState<Submission[]>([])
+  const [filterValues, setFilterValues] = useState<string[]>([])
+
+  function toggleFilterValue(filter: string): void {
```

Comment:
What if we move the state of the filter and this logic inside the `Filter` component?